docs/advanced/debugging.rst

```diff
@@ -27,15 +27,335 @@ And, of course, if you're hitting that ever-challenging ``No scope with a Tag ma
 Symbols and Sources
 -------------------
 
-Autofac publishes symbols and sources for its various packages on SymbolSource and MyGet. (Older packages are on SymbolSource, newer stuff has moved to MyGet.)
+Autofac packages have been updated `to use Source Link <https://github.com/dotnet/sourcelink>`_ so the debugging experience right to the source should feel native.
+
+Older packages and packages that haven't been updated yet have symbols published to MyGet and SymbolSource.
 
 You can set up Visual Studio to debug/step *right into Autofac source* using the following symbol servers:
 
 - ``https://www.myget.org/F/autofac/symbols/``
 - ``http://srv.symbolsource.org/pdb/Public/``
 
-`There is documentation on MyGet explaining how to configure Visual Studio to make this work. <http://docs.myget.org/docs/reference/symbolsource>`_
+`There is documentation on MyGet explaining how to configure Visual Studio to make symbol servers work work. <http://docs.myget.org/docs/reference/symbolsource>`_
+
+Diagnostics
+-----------
+
+Autofac 6.0 introduced diagnostics support in the form of `System.Diagnostics.DiagnosticSource <https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.diagnosticsource?view=netcore-3.1>`_. This allows you to intercept diagnostic events from Autofac.
+
+.. note::
+
+    Diagnostics aren't free. You will get better performance if you **don't** have a diagnostic listener attached to the container. Further, tracers like ``DefaultDiagnosticTracer`` that generate a full trace of an operation will increase memory and resource usage as they have to hold onto data during the entire resolve operation in order to generate a complete trace. It is recommended you only use diagnostics in a non-production environment; or use diagnostics listeners that handle individual events without tracking full operations.
```

Comment:
Do you have line wrapping on in VS Code? Because this a looong line. Can I request line breaks in the rst files?

Reply:
I'm really torn on line wrapping in docs. Like, writing it the first time, awesome, it's wrapped. Editing it later? Re-wrapping usually ends up being a pain in the ass.

You guys didn't like my "use `this` so it's easier to read outside a code editor" thing... :eyes:

Reply:
I mean, touché. I'll pick my battles on this one and leave it....